client/rpc-servers/src/lib.rs

```diff
@@ -64,98 +66,51 @@ pub struct WsConfig {
-	metrics: Option<RpcMetrics>,
-	rpc_api: RpcModule<M>,
-	rt: tokio::runtime::Handle,
+/// Start RPC server listening on given address.
+pub async fn start_server<M: Send + Sync + 'static>(
+	config: Config<'_, M>,
 ) -> Result<ServerHandle, Box<dyn StdError + Send + Sync>> {
-	let max_payload_in = payload_size_or_default(max_payload_in_mb) as u32;
-	let max_payload_out = payload_size_or_default(max_payload_out_mb) as u32;
-	let host_filter = hosts_filter(cors.is_some(), &addrs);
-
-	let middleware = tower::ServiceBuilder::new()
-		// Proxy `GET /health` requests to internal `system_health` method.
-		.layer(ProxyGetRequestLayer::new("/health", "system_health")?)
-		.layer(try_into_cors(cors)?);
-
-	let builder = ServerBuilder::new()
-		.max_request_body_size(max_payload_in)
-		.max_response_body_size(max_payload_out)
-		.set_host_filtering(host_filter)
-		.set_middleware(middleware)
-		.custom_tokio_runtime(rt)
-		.http_only();
-
-	let rpc_api = build_rpc_api(rpc_api);
-	let (handle, addr) = if let Some(metrics) = metrics {
-		let server = builder.set_logger(metrics).build(&addrs[..]).await?;
-		let addr = server.local_addr();
-		(server.start(rpc_api)?, addr)
-	} else {
-		let server = builder.build(&addrs[..]).await?;
-		let addr = server.local_addr();
-		(server.start(rpc_api)?, addr)
-	};
-
-	log::info!(
-		"Running JSON-RPC HTTP server: addr={}, allowed origins={}",
-		addr.map_or_else(|_| "unknown".to_string(), |a| a.to_string()),
-		format_cors(cors)
-	);
-
-	Ok(handle)
-}
-
-/// Start a JSON-RPC server listening on given address that supports both HTTP and WS.
-pub async fn start<M: Send + Sync + 'static>(
-	addrs: [SocketAddr; 2],
-	cors: Option<&Vec<String>>,
-	ws_config: WsConfig,
-	metrics: Option<RpcMetrics>,
-	rpc_api: RpcModule<M>,
-	rt: tokio::runtime::Handle,
-	id_provider: Option<Box<dyn IdProvider>>,
-) -> Result<ServerHandle, Box<dyn StdError + Send + Sync>> {
-	let (max_payload_in, max_payload_out, max_connections, max_subs_per_conn) =
-		ws_config.deconstruct();
-
-	let host_filter = hosts_filter(cors.is_some(), &addrs);
+	let Config {
+		addrs,
+		cors,
+		max_payload_in_mb,
+		max_payload_out_mb,
+		max_connections,
+		max_subs_per_conn,
+		metrics,
+		id_provider,
+		tokio_handle,
+		rpc_api,
+	} = config;
+
+	let host_filter = hosts_filtering(cors.is_some(), &addrs);
 
 	let middleware = tower::ServiceBuilder::new()
 		// Proxy `GET /health` requests to internal `system_health` method.
 		.layer(ProxyGetRequestLayer::new("/health", "system_health")?)
 		.layer(try_into_cors(cors)?);
 
 	let mut builder = ServerBuilder::new()
-		.max_request_body_size(max_payload_in)
-		.max_response_body_size(max_payload_out)
-		.max_connections(max_connections)
-		.max_subscriptions_per_connection(max_subs_per_conn)
+		.max_request_body_size(payload_size_or_default(max_payload_in_mb))
+		.max_response_body_size(payload_size_or_default(max_payload_out_mb))
+		.max_connections(max_connections.map_or(RPC_MAX_CONNECTIONS, |m| m as u32))
```

Comment:
this is confusing.. we say earlier that "default is None (unlimited)" number of connections, but here we seem to restrict the number of connections to 100

Reply:
Earlier we did only `restrict ws connections to 100 by default` and the http server had "unlimited allowed connections" 

After this PR there is not a separate ws and http server so I thought it was most reasonable to restrict it to 100 by default but I understand your point.

Do you prefer something else?

Reply:
Should we update the docs to reflect the new default (100)?

Reply:
Sounds good to me.

What do you reckon changing the `structopt CLI Option<usize>` to `usize` and provide a default value?
Probably the easiest to maintain in the long run

Reply:
> What do you think about changing the structopt CLI Option<usize> to usize and provide a default value?
Then we don't really have be bother unwrap_or in the rpc code itself.

I'm in favor